src/sign/non_interactive_sign/share.rs

```diff
@@ -49,3 +49,17 @@ impl std::ops::Add<SignatureShare> for Scalar {
         self + rhs.0
     }
 }
+
+impl std::ops::Mul<SignatureShare> for SignatureShare {
```

Comment:
I'm not sure where this is coming from or why it's needed from the PR. Signatures shares for supporting t out of t should be the same as n out of n (you just need to sum). This is related to supporting the HD wallet, correct?

Reply:
This is indeed not needed. I just removed it from the code base.